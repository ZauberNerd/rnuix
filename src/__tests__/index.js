import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import createApp from '../';

const renderer = new ShallowRenderer();

describe('App', () => {
    it('renders without demos', () => {
        const App = createApp();
        const tree = renderer.render(<App />);
        expect(tree).toMatchSnapshot();
    });

    it('renders with demos', () => {
        const render = jest.fn();
        const components = [
            {
                displayName: 'DisplayName 1',
                description: 'Description 1',
                demos: [
                    {
                        title: 'Demo 1a',
                        render,
                    },
                    {
                        title: 'Demo 1b',
                        render,
                    },
                ],
            },
            {
                displayName: 'DisplayName 2',
                description: 'Description 2',
                demos: [
                    {
                        title: 'Demo 2a',
                        render,
                    },
                    {
                        title: 'Demo 2b',
                        render,
                    },
                ],
            },
        ];
        const App = createApp(components);
        const tree = renderer.render(<App />);
        expect(tree).toMatchSnapshot();
    });
});
