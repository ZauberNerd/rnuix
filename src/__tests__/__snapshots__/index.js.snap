// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders with demos 1`] = `
<NavigationContainer
  screenProps={
    Object {
      "components": Array [
        Object {
          "demos": Array [
            Object {
              "render": [Function],
              "title": "Demo 1a",
            },
            Object {
              "render": [Function],
              "title": "Demo 1b",
            },
          ],
          "description": "Description 1",
          "displayName": "DisplayName 1",
        },
        Object {
          "demos": Array [
            Object {
              "render": [Function],
              "title": "Demo 2a",
            },
            Object {
              "render": [Function],
              "title": "Demo 2b",
            },
          ],
          "description": "Description 2",
          "displayName": "DisplayName 2",
        },
      ],
    }
  }
/>
`;

exports[`App renders without demos 1`] = `
<NavigationContainer
  screenProps={
    Object {
      "components": Array [],
    }
  }
/>
`;
